libc/src/__support/block.h

```diff
@@ -338,32 +323,26 @@ class Block {
   /// ensure the split will succeed.
   static Block *split_impl(Block *&block, size_t new_inner_size);
 
-  /// Offset (in increments of the minimum alignment) from this block to the
-  /// previous block. 0 if this is the first block.
+  /// Offset from this block to the previous block. 0 if this is the first
+  /// block.
   offset_type prev_ = 0;
 
-  /// Offset (in increments of the minimum alignment) from this block to the
-  /// next block. Valid even if this is the last block, since it equals the
-  /// size of the block.
+  /// Offset from this block to the next block. Valid even if this is the last
+  /// block, since it equals the size of the block.
   offset_type next_ = 0;
```

Comment:
Since `next_` now encodes the old `info_` struct, do you think it would be more readable to use bitfields rather than depend on masks? Something like

```
    struct {
        offset_type next : sizeof(offset_type) * CHAR_BIT - 2;
        unsigned int used : 1;
        unsigned int last : 1;
    } info_;

size_t outer_size() const { return info_.next; }
void clear_last() { info_.last_ = 0; }
```

Other than this, LGTM

Reply:
I thought about doing this, but I couldn't find any standard that guarantees the order of bitfields on all the targets libc supports or will eventually support. It wouldn't be the end of the world if the flags were in the high order bits, but then extracting the size would be a mask and a shift, rather than just a mask.

Leaving open to see if anyone knows of a better way to express this.